src/lib/pdfconverter.cc

```diff
@@ -546,13 +546,26 @@ void PdfConverterPrivate::findLinks(QWebFrame * frame, QVector<QPair<QWebElement
 		uexternal  = PageObject::webPageToObject[frame->page()]->settings.useExternalLinks;
 	}
 	if (!ulocal && !uexternal) return;
-	foreach (const QWebElement & elm, frame->findAllElements("a")) {
+	foreach (QWebElement elm, frame->findAllElements("a")) {
 		QString n=elm.attribute("name");
 		if (n.isEmpty()) n=elm.attribute("ns0:name");
 		if (n.startsWith("__WKANCHOR_")) anchors[n] = elm;
 
 		QString h=elm.attribute("href");
-		if (h.isEmpty()) h=elm.attribute("ns0:href");
+		if (h.isEmpty()) {
+			h=elm.attribute("ns0:href");
+
+			//Internal links (anchor tags) are not linked correctly
+			//when the anchor's content is empty. This adds a space with
+			//a height of 0 when it is empty so that internal links
+			//are linked correctly.
+			if(elm.firstChild().isNull()){
```

Comment:
The check could be changed to also check that the ```a``` tag has an ```id``` or ```name``` too. That would be a stronger check and less likely to produce side effects. However an empty ```href``` attribute is the key condition.

For ```ns0:href```, are you saying that we also need to check that ```ns0:href``` is empty?